csrc/fusion.cpp

```diff
@@ -797,28 +797,37 @@ void Fusion::aliasOutputToInput(
     }
   }
 
-  NVF_ERROR(
-      isAliasCompatible(input, output),
-      "The input and output values are not alias-compatible.");
-  // Let integration hide any output that wasn't a fusion output when
-  // `aliasOutputToInput` was called. For example, running mean and var for
-  // batch norm.
-  io_alias_[output] = AliasInfo{
-      .type = type,
-      .aliased_io = input,
-      .hide_output = !output->isFusionOutput()};
+  // AllocationType::Evaluate
+  if (!input) { // Non-aliased outputs marked for expression evaluator.
```

Comment:
```suggestion
  if (input == nullptr) { // Non-aliased outputs marked for expression evaluator, e.g., a matmul running in fallback mode.
```
so it's immediately clear from this line that we are checking a pointer not a boolean value.